src/components/SetupNFTClaimer.vue

```diff
@@ -0,0 +1,269 @@
+          <BaseMessage level="info" class="mb-3">
+            Updates will only be applied to future proposals
+          </BaseMessage>
+          <TuneInput
+            v-model="input.maxSupply"
+            label="Max supply"
+            hint="Maximum number of NFTs per proposal"
+            placeholder="100"
+            min="1"
+            type="number"
+            :disabled="isViewOnly"
+            :error="validationErrors?.maxSupply"
+            autofocus
+            @update:model-value="value => (input.maxSupply = Number(value))"
+          />
+
+          <TuneInput
+            v-model="input.formattedMintPrice"
+            label="Mint price"
+            min="0"
+            :hint="`The mint price for each NFT, in ${MINT_CURRENCY}`"
+            type="number"
+            placeholder="0.5"
+            :error="validationErrors?.formattedMintPrice"
+            :disabled="isViewOnly"
+            @update:model-value="
+              value => (input.formattedMintPrice = Number(value))
+            "
+          />
+
+          <TuneInput
+            v-model="input.proposerFee"
+            label="Proposer fees"
+            type="number"
+            hint="Percentage of the mint price shared with the proposal author"
+            placeholder="5"
+            min="0"
+            :max="maxProposerCut"
+            :error="validationErrors?.proposerFee"
+            :disabled="isViewOnly"
+            @update:model-value="value => (input.proposerFee = Number(value))"
+          />
+
+          <TuneInput
+            v-model.trim="input.treasuryAddress"
+            label="Space treasury wallet"
+            hint="Wallet address receiving the funds from the mint"
+            placeholder="0x0000"
+            :error="validationErrors?.treasuryAddress"
+            :disabled="isViewOnly"
+          />
+
+          <div v-if="context !== 'settings'" class="px-4 md:px-0">
+            <SetupButtonBack @click="emit('back')" />
+            <SetupButtonNext @click="nextStep" />
+          </div>
+        </div>
+      </BaseBlock>
+
+      <div v-if="!contractInfo" class="flex gap-5 px-4 pt-2 md:px-0">
+        <BaseButton
+          primary
+          class="w-full"
+          :disabled="isViewOnly"
+          :loading="loading"
+          type="submit"
+          @click="submit"
+        >
+          Setup SnapIt!
+        </BaseButton>
+      </div>
+      <div v-else class="flex gap-5 px-4 pt-2 md:px-0">
+        <BaseButton
+          class="mb-2 block w-full"
+          :disabled="isViewOnly || !dirtyFields"
+          @click="resetForm"
+        >
+          {{ $t('reset') }}
+        </BaseButton>
+        <BaseButton
+          primary
+          class="block w-full"
+          type="submit"
+          :disabled="
+            isViewOnly ||
+            Object.keys(validationErrors).length > 0 ||
+            !dirtyFields
+          "
+          :loading="loading"
+          @click="submit"
+        >
+          Save
```

Comment:
What do you mean ? there is no modal. 

And save button remain disabled until there are changes in the form. We could make it always enabled, and transparently handled the submit flow behind the scene (do nothing when nothings' changed, and do not trigger the tx)

Reply:
SAVE button is now always enabled, and a flash notification will be shown when trying to save the same data

Reply:
> What do you mean ? there is no modal.

The modal I described in my previous comment,

Reply:
> What happens if I save before the transaction is confirmed?

You can not SAVE before the transaction is confirmed, because we show the form only once confirmed